lib/datadog/core/environment/thread_count.rb

```diff
@@ -0,0 +1,18 @@
+module Datadog
+  module Core
+    module Environment
+      # Retrieves number of threads from runtime
+      module ThreadCount
+        module_function
+
+        def value
+          Thread.list.count
+        end
+
+        def available?
+          Thread.respond_to?(:list)
+        end
```

Comment:
We should remove this, as `Thread#list` has existed [at least since 1.9](https://ruby-doc.org/core-1.9.1/Thread.html#method-c-list). Not for this PR, of course, but can I convince you to pick it up afterwards? ;)

Reply:
If you send me another funny video, consider me persuaded.

Reply:
![](https://media.giphy.com/media/98qUQf8dzAGe4/giphy.gif)